shell/platform/android/io/flutter/plugin/common/MethodCall.java

```diff
@@ -0,0 +1,29 @@
+package io.flutter.plugin.common;
+
+import java.util.Objects;
+
+/**
+ * Command object representing a method call on a {@link FlutterChannel}.
+ */
+public final class MethodCall {
+    /**
+     * The name of the called method.
+     */
+    public final String method;
+
+    /**
+     * Arguments for the call.
+     */
+    public final Object arguments;
```

Comment:
Ok.  I think the method call stuff isn't layered in as well as it could be.  You've wired this type into the codec base class, which is why it needs to be so generic.  If you wired it in to a more specific class (e.g., a subclass of the JSONCodec that understands the particular JSON schema we use for method calls), then you can use a less generic type without foreclosing possibilities like the one you mention here.